training/src/main/java/wolox/training/models/Book.java

```diff
@@ -135,4 +136,28 @@ public void setGenre(String genre) {
     public void setClients(List<Client> clients) {
         this.clients = clients;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof Book)) return false;
+        Book book = (Book) o;
+        return id == book.id &&
```

Comment:
This is tricky as it depends on the use case. But normally, two books will be the same if everything else matches, although the ID might not match (i.e. a persited version of the book, against a not-persisted version). Take that into account.